src/Snackbar/index.tsx

```diff
@@ -0,0 +1,205 @@
+import { createPortal } from 'react-dom';
+import theme from 'src/styles/theme';
+import { generateAnchorOrigin, MESSAGE_TYPES, SnackbarCloseReason, SnackbarOrigin } from './Snackbar.util';
+import SnackbarContent from './SnackbarContent';
+import { SnackbarWrapper, StyledSnackbarCloseButton, TRANSITION_CLOSE_TIME } from './snackbarStyles';
+import { ReactComponent as CloseIcon } from 'src/Snackbar/assets/closeIcon.svg';
+
+export interface SnackbarProps {
+    /**
+     * The action to display. It renders after the message, at the end of the snackbar.
+     */
+    action?: ReactNode;
+    /**
+     * The anchor of the `Snackbar`.
+     */
+    anchorOrigin?: SnackbarOrigin;
+    /**
+     * The number of milliseconds to wait before automatically calling the
+     * `onClose` function. `onClose` should then set the state of the `open`
+     * prop to hide the Snackbar. This behavior is disabled by default with
+     * the `null` value. If left undefined will auto hide after 4000ms.
+     */
+    autoHideDuration?: number | null;
+    /**
+     * The message to display.
+     */
+    message?: string;
+    /**
+     * Callback fired when the component requests to be closed.
+     * Typically `onClose` is used to set state in the parent component,
+     * which is used to control the `Snackbar` `open` prop.
+     * The `reason` parameter can optionally be used to control the response to `onClose`,
+     * for example ignoring `clickaway`.
+     *
+     * @param {string} reason Can be: `"timeout"` (`autoHideDuration` expired), `"clickaway"`.
+     */
+    onClose?: (reason: SnackbarCloseReason) => void;
+    /**
+     * If `true`, `Snackbar` is open.
+     */
+    open?: boolean;
+    /**
+     * Message types used to determine icon color and icon to render.
+     */
+    type: MESSAGE_TYPES;
+    /**
+     * When true, the component will mount a div to the body and render the popover through it.
+     * This is useful when the popover would be inside a scrollable container or one with "overflow: hidden"
+     * so it doesn't get cut off. Uses IntersectionObserver and needs a polyfill if IE compatibility is needed.
+     */
+    renderInPortal?: boolean;
+}
+
+/**
+ * Snackbar Component
+ *
+ * The Snackbar component is a UI informational overlay.
+ * This can be used to display messages, as well as provide an action with an allocated type that has it's own corresponding icon and styling.
+ * The state is not managed inside this component and visibility (via the `open` prop) needs to be maintained in the parent component.
+ * The `renderInPortal` prop can be used to append a div to the body.
+ *
+ */
+const Snackbar: FC<SnackbarProps> = ({
+    anchorOrigin = { vertical: 'bottom', horizontal: 'left' },
+    open,
+    autoHideDuration = 4000,
+    onClose,
+    message,
+    type = MESSAGE_TYPES.INFO,
+    renderInPortal = false,
+    
+    action = <StyledSnackbarCloseButton
```

Comment:
Thanks for adding the default!

nit: This is kinda hard to read. Possible ways to fix are either better format with the spacing/indents or create a utility function and do something like:

```tsx
// in util file
const createDefaultAction = (onClose: (reason: SnackbarCloseReason) => void) => {
    return (
        <StyledSnackbarCloseButton
            alternate
            className='closeIcon'
            key='close'
            aria-label='Close'
            onClick={() => onClose ? onClose('timeout') : undefined}
            icon={<CloseIcon />}
        />
    );
};


// back in this file
action = createDefaultAction(onClose)
```